src/materialized/src/version_check.rs

```diff
@@ -7,30 +7,33 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use std::time::Duration;
+use std::collections::HashSet;
+use std::time::{Duration, Instant};
 
 use anyhow::bail;
 use log::{debug, log, Level};
 use semver::{Identifier, Version};
 use serde::{Deserialize, Serialize};
+use tokio::time::{sleep as tokio_sleep, Duration as TokioDuration};
 
 use ore::retry::RetryBuilder;
 
+use crate::server_metrics::{
+    filter_metrics, load_prom_metrics, METRIC_SERVER_METADATA, METRIC_WORKER_COUNT,
+};
 use crate::BUILD_INFO;
 
+/// How often we report telemetry
+const TELEMETRY_FREQUENCY_SECS: u64 = 3600;
```

Comment:
`Duration::from_secs` is a constant function, so recommend making this constant of type `Duration` so that you don't need to encode its units in its name:

```
const TELEMETRY_FREQUENCY: Duration = Duration::from_secs(3600)
```